2022-11-27.rb

```diff
@@ -0,0 +1,60 @@
+require File.expand_path(File.dirname(__FILE__) + '/assert')
+require 'pry'
+require 'json'
+
+# Level: Medium
+# This problem was asked by Google.
+
+# Given the root to a binary tree, implement serialize(root), which serializes the tree into a string,
+# and deserialize(s), which deserializes the string back into the tree.
+# 
+# For example, given the following Node class:
+# 
+# class Node:
+#     def __init__(self, val, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+#
+# The following test should pass:
+# 
+# node = Node('root', Node('left', Node('left.left')), Node('right'))
+# assert deserialize(serialize(node)).left.left.val == 'left.left'
+
+# my Ruby implementation of the Node class given in the instructions in Python
+Node = Struct.new(:val, :left, :right)
+
+######### SOLUTIONS #########
+
+# I'm serializing to JSON, since that's a binary tree serialized format I know, and I can use the JSON parser for the deserialization
+# I'm of course using recursivity, and for such tasks I'm not sure there's any other/better way
```

Comment:
not a ruby expert but could you end up with stack overflow using recursivity with very large tree?

Another way would be to use a queue or stack